src/components/pages/auth/login.tsx

```diff
@@ -147,9 +147,9 @@ export function Login({ oauth }: { oauth?: boolean }) {
 
   async function onLogin(values: SubmitSelfServiceLoginFlowBody) {
     if (!flow?.id) return
-    await router.push(`${router.pathname}?flow=${flow.id}`, undefined, {
-      shallow: true,
-    })
+
+    router.query.flow = flow.id
+    await router.push(router)
```

Comment:
@hugotiburtino not sure why pushing router to router works and what it should do. Could you explain or send  a docs link?

Reply:
I was also suprised that it works. Based on https://stackoverflow.com/questions/61466891/nextjs-appending-a-query-param-to-a-dynamic-route

Reply:
But what is the difference to the current code (besides the routing not being shallow)?

Reply:
It preserves the other query params that the url may have. In the specific case of the OAuth path it is important to maintain the login_challenge query parameter in order to solve the bug of the linked issue.

Reply:
We could alternately not change the url at all and if the user refreshes the page they lose their inputs. At the end of the day it is maybe what they want

Reply:
yeah, I think for login we should try that for now.
Maybe it's different for e.g. registering .

Reply:
Sorry, I have to correct myself: the recovery of data based on flow id is not implemented, Indeed we get the data from backend, but we don't fill in the fields with them So, the way it is now the user gets a very bad experience after refresh, 

See here in registration: error message with empty field
![image](https://user-images.githubusercontent.com/45924645/206875093-c81845f1-0841-4f18-8842-7dae9971d6df.png)

I think it is safe to delete these lines in login and in registration (I'll test with other flows and if I see that there isn't problem also you will get my commits on that)

Reply:
> I think it is safe to delete these lines in login and in registration (I'll test with other flows and if I see that there isn't problem also you will get my commits on that)

Of course, unless you think we should make effort to implement it now...

Otherwise, such lines are not only useless but also reason for bad UX.